src/transformers/cache_utils.py

```diff
@@ -0,0 +1,157 @@
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional, Tuple, TypeVar
+
+import torch
+
+
+T = TypeVar("T")
+
+
+class Cache(ABC):
```

Comment:
```suggestion
class Cache(ABC):
```

(nit) Do we really need the ABC abstraction? I don't think this does anything and don't see the reason to add it.

Reply:
We don't need it, no. The reason I've added it is to avoid people from ever trying to initialize `Cache` rather than `DynamicCache`, but this can just be avoided by not importing `Cache` into `src/transformers/__init__.py`, i.e. preventing `from tranformers import Cache` from working.

@gante do you want to implement this in your work in https://github.com/tomaarsen/transformers/pull/1?